rs/monitoring/metrics/src/adapter_metrics_registry.rs

```diff
@@ -54,24 +53,22 @@ impl AdapterMetricsRegistry {
     /// Write accesses to here can not be starved by `gather()` calls, due to
     /// the write-preffering behaviour of the used `tokio::sync::RwLock`.
     pub fn register(&self, adapter_metrics: AdapterMetrics) -> Result<(), Error> {
-        if self
-            .adapters
-            .blocking_read()
+        let mut adapters = self.adapters.write();
+        if adapters
             .iter()
             .any(|a| a.get_name() == adapter_metrics.get_name())
         {
             return Err(Error::AlreadyReg);
         }
-        self.adapters.blocking_write().push(adapter_metrics);
+        adapters.push(adapter_metrics);
         Ok(())
     }
 
     /// Concurrently scrapes metrics from all registered adapters.
     pub async fn gather(&self, timeout: Duration) -> Vec<MetricFamily> {
+        let adapters = self.adapters.read().clone();
```

Comment:
FYI, `AdapterMetrics::name` could be a `&'static str`. That would make it even cheaper to clone.